File: hw/chavez/u3/hw24CompositePaterns/Python/model/supervisor.py
from model.employee import Employee


class Supervisor(Employee):

    def __init__(self, name, title):
        super().__init__(name, title)
        self.direct_reports = []

    def add(self, employee):
        self.direct_reports.append(employee)

    def state_name(self):
        super().state_name()

        for employee in self.direct_reports:
            employee.state_name()

File: hw/tipan/u3/hw25CompositePattern/CompositePatternJava/CompositePattern/src/main/java/ec/edu/espe/compositepattern/model/Supervisor.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.compositepattern.model;

import java.util.ArrayList;
import java.util.List;

public abstract class Supervisor extends Employee {

    protected List<Employee> directReports;

    public Supervisor() {
        directReports = new ArrayList<>();
    }

    public Supervisor(String name, String title) {
        super(name, title);
        directReports = new ArrayList<>();
    }

    public void add(Employee employee) {
        directReports.add(employee);
    }

    public List<Employee> getDirectReports() {
        return directReports;
    }

    @Override
    public void stateName() {

        super.stateName();

        for (Employee employee : directReports) {
            employee.stateName();
        }

    }

}